amundsen_application/static/js/components/common/Breadcrumb/index.tsx

```diff
@@ -23,7 +28,20 @@ export const Breadcrumb: React.SFC<BreadcrumbProps> = (props) => {
     path = '/';
     text = 'Home';
     if (props.searchTerm) {
-      path = `/search?searchTerm=${props.searchTerm}&selectedTab=table&pageIndex=0`;
+      let index = 0;
+      switch (props.selectedTab) {
+        case ResourceType.table:
+          index = props.tableIndex;
+          break;
+        case ResourceType.dashboard:
+          index = props.dashboardIndex;
+          break;
+        case ResourceType.user:
+          index = props.userIndex;
+          break;
+      }
+
+      path = `/search?searchTerm=${props.searchTerm}&selectedTab=${props.selectedTab}&pageIndex=${index}`;
```

Comment:
In addition, if we do decide to explore this alternative implementation I'd be interested in there being a commit/series of commits where we can look at the necessary changes without any adhoc work -- unless that adhoc work is needed for a cleaner implementation. A lot cleanup and refactor has been done in the past couple weeks/months as myself, you and Ryan all worked on different things and I'm concerned about more bugs slipping through in reviews. It would be great if we could finally cut a release soon that doesn't have any bugs or "unintended features" before H2 work begins.

Reply:
I like the idea of simplifying the path to just say `use last search` rather than encoding the full path inside this component. 

For changes in search page logic changes, much of the complexity comes from trying to sync 3 distinct entities: 1. global state, 2. browser URL, and 3. component state. Removing 3 enables us to simplify that logic and will be easier to maintain and have fewer bugs.